### ISSUE TITLE
Minutes: SCP Updates [every Monday]

### ISSUE THREAD

**Issue by moul** [gnolang/meetings#39]
We'll set up recurring meetings with SCP program members, likely on a weekly or bi-weekly basis.

The goal will be for members to fill a live HackMD document, then discuss the contents with some core team members.

This approach, similar to issue gnolang/gno#36, will help track recent updates by simply copying the HackMD file into a new comment.

cc @VT-Cosmos @leohhhn

**Comment by leohhhn** [comment]
Week 17-21 June 2024:

### Alex (@agherasie)

#### What did I work on last week?
- Read docs, explored realms and followed how-to-guides
- Fixed some minor things in the docs
- Built a small project aiming to be a decentralised https://www.futureme.org/ 
    - inspired by r/demo/memeland
    - smart contract for posting letters and reading them depending on date
    - react frontend (kinda ugly because I wanted to try out tailwind, I'm used to ChakraUI)

#### What will I focus on this week?
- Finishing the small project mentioned above (adding a feature suggested by Reza)
- Writing more gno code, now that I am satisfied with how much of the docs I've read

#### Do I have any blockers?
- Not really

---

### Mohammed (@Molaryy)

#### What did I work on last week?
- I finished all the how-to-guides
- Did some PRs to fix some little things in the docs
- Started thinking about an opensea like using the GRC721

#### What will I focus on this week?
- I will mainly focus on the opensea like with Alex
- We are thinking about an oracle in gno and how to implement it

#### Do I have any blockers?
- Not really

---
### Reza (@RezaRahemtola)

#### What did I work on last week?
- Finished reading and going through the docs / examples / guides
- Did some PRs on the documentation
- Opened some issues related to docs / SDKs

#### What will I focus on this week?
- Start building small projects / writing my own Gno code
- Thinking about a habit builder with locked tokens
- Looking into other ideas (Aave-like, multisig)

#### Do I have any blockers?
- Not really

**Comment by leohhhn** [comment]
Week 24-28 June 2024:

### Alex (@agherasie)

#### What did I work on last week?
- [r/demo/futuregno](https://github.com/agherasie/futuregno) (a very small web3 version of [futureMe](https://www.futureme.org/) that I built for learning purposes)
- Reading various topics on blockchain / web3
- With @molaryy, started working on [OpenOcean](https://github.com/Molaryy/openocean), our own [OpenSea](https://opensea.io/) also for learning purposes, but this time with prettier UI

#### What will I focus on this week?
- Continuing work on OpenOcean alongside @molaryy
- Try to make another contribution to the gno codebase

#### Do I have any blockers?
- Not sure if there is a way to encrypt the contents of posts in `futuregno` so that the content of a post is not available to people other than the writer (and only after the delivery date)

---
### Reza (@RezaRahemtola)

#### What did I work on last week?
- Exploring and testing some things with Adena and gno-js. Created some issues and PRs related to this:
  -  https://github.com/gnolang/gno-js-client/issues/127
  -  https://github.com/harry-hov/vscode-gno/pull/5
  -  https://github.com/gnolang/gno/pull/2419
  -  https://github.com/gnolang/gno-js-client/issues/128
  -  https://github.com/gnolang/gno/issues/2420 
- Starting working on gnovox, a Medium-like inspired from the Gno blog (but blocked/slowed down by the issues I encountered)
- Waiting on https://github.com/gnolang/gno/pull/2307 to be merged to maybe start my habit builder project

#### What will I focus on this week?
- Gnovox, will make a MVP version (create blog, post articles...) and then I'll see if it's interesting to continue and to build a real project

#### Do I have any blockers?
- Not really (issues already on github with links above)

---
### Lenny (@lennyvong)

#### What did I work on last week?
- Did some research about some good projects and find two i would like to dig.
    - A decentralize multi-sig based on aleph for the "backend". Similar to safe on ethereum but using Aleph.im which is a decentralize plateform to post data on it.
    - A request-reponse Oracle.
    - VC plateform maybe using token streaming.

#### What will I focus on this week?
- This week I will focus myself on the multi-sig i think its a good start and a good first project.
- I will discuss about Gnorkle because it is a attempt to a generic Oracle but idk how far this attempt works or not. So I need to make some research.

#### Do I have any blockers?
- About Gnorkle I need some info about.
- Need info about token streaming.

---

### Mohammed (@Molaryy)

#### What did I work on last week?
- I started a project with @agherasie, [openocean](https://github.com/Molaryy/openocean). It's an opensea like, Alex finished the frontend.

#### What will I focus on this week?
- I will mainly finish openocean and then we will find another project to do, in group or solo.

#### Do I have any blockers?
- I got a few blockers creating my own NFT and there are not a lot of examples but I'm trying to see what can I do.

**Comment by leohhhn** [comment]
July 1 - 5 2024
### Alexandru (@agherasie)

#### What did I work on last week?
- Opened two PRs: 
    - [fixing the footer display](https://github.com/gnolang/gno/pull/2462)
    - [adding link to readme if present in the package on gno.land](https://github.com/gnolang/gno/pull/2461)
- Read about web3 in general, 'lending' in particular because I will be working on an AAVE-like with @lennyvong

#### What will I focus on this week?
- Working on projects: openocean with @molaryy and the AAVE-like with @lennyvong and @RezaRahemtola

#### Do I have any blockers?
- Not exactly, I had a little difficulty thinking of what project to do last week but we ended up coming up with something with @lennyvong over the weekend

### Lenny Vongphouthone (@lennyvong)

#### What did I work on last week?
- Oracle
    - Sent a message to Matteo to talk about Gnorkle. Need to contact the developer of Gnorkle to get more info.
- Multi-sig
    - Made some research about the actual multi-sig by gnokey with @agherasie. It seems to be functionnal but we don't really succeed to sue it.
- Lending Plateform (Gnoove)
    - I had the idea to build a lending plateform with agherasie and @reza. And we started to setup the project and made some research about the lending in general.

#### What will I focus on this week?
- Focus myself on a PoC of the lending plateform. And continue to make some research about gnorkle btw.

#### Do I have any blockers?
- About the multi-sig bc at the beggining with @agherasie we started the multi-sig but apparently it could be useless so we switch it up.
- Gnorkle, complicate to get some infos.
- Not really about the lending plateforme

### Reza (@RezaRahemtola)

#### What did I work on last week?
- Made an MVP of gnovox (the Medium-like platform): https://github.com/RezaRahemtola/gnovox
- Found some issues / PRS around gno-js and Adena:
  - https://github.com/onbloc/adena-wallet/issues/525
  - https://github.com/onbloc/adena-wallet/issues/526
  - https://github.com/gnolang/docs.gno.land/issues/49 

#### What will I focus on this week?
- Make gnovox really usable (UI, error checks, edit posts)
- Help on the lending platform

#### Do I have any blockers?
- Not really, storage in smart contracts isn't really intuitive, 
  looking forward to SQL-like interfaces

---

### Mohammed (@Molaryy)

#### What did I work on last week?
- Implemented the storage of images into IPFS
- Implemented the realm to storage all the NFT

#### What will I focus on this week?
- The compatibility of the real and the frontend of [openocean](https://github.com/Molaryy/openocean)
- Join Alexandru and Lenny for their AAVE-like

#### Do I have any blockers?
- Not at the moment, but I struggled implementing an avl.Tree after following the docs.

**Comment by leohhhn** [comment]
8-12 July 2024

### Alex (@agherasie)

#### What did I work on last week?
- We tried to look into disperse with Lenny, but we were blocked because of an issue
- I built another small project in gno, [gno-forms](https://github.com/agherasie/gno-forms) which is a web3 version of Google Forms, which was a good experience to get more used to writing gno code

#### What will I focus on this week?
- I'd like to make a few adjustments to my gno-forms project and also work on disperse and maybe the AAVE-like project we mentioned with Lenny and Reza

#### Do I have any blockers?
- We asked a question on disperse on the discord but we didn't get an answer yet, about how we could send multiple coins using --send.
- Also, this brings me to another question I had: can you send arrays as arguments to gno function calls ? or is the best way to send JSON strings instead ?

---


### Mohammed (@Molaryy)

#### What did I work on last week?
- I worked in the realm of openocean, I need a small fixes and it will be ready

#### What will I focus on this week?
- Finish openocean and start a new PoC
- I will also join Alex and Lenny in their project

#### Do I have any blockers?
- Not at the moment

---

### Reza (@RezaRahemtola)

#### What did I work on last week?
- Completing the MVP of gnovox: https://github.com/RezaRahemtola/gnovox, which can be useful as an inspiration for people you want to start small projects using Gno / JS libraries and SDKs
- Learning more about Coins / GRC-20 in Gno
- Learning more about Aave and lending protocols in general (discussed with Marc Zeller IRL), started to think about the architecture of a similar protocol in Gno 

#### What will I focus on this week?
- Help finishing gno disperse if needed
- Start working on an MVP of an "Aave" like in Gno
- Discuss with the others what we want to do next / maybe start a real and more long term project

#### Do I have any blockers?
- Not right now, will probably send messages in the discord during the week


@lennyvong

No updates posted.

**Comment by leohhhn** [comment]
July 15-19th 2024

### Lenny (@lennyvong)

#### What did I work on last week?
- Essentially work on disperse with @agherasie.
Globally everything about the dapp works but got problems about tests.
- Spot some differences in the doc about grc20 and one error about std.GetOrigCaller -> need to do some PRs.

#### What will I focus on this week?
- Finish disperse with @agherasie.
- Start a new personal side project.
- Talk about Lending project and maybe start a PoC or sum.

#### Do I have any blockers?
- About building some tests for disperse. I think i'm using the Approve and Allowance about GRC20 proprely but I get some weird stuff. In the function disperseToken I'm using TransferFrom.

---

 
### Alex (@agherasie)

#### What did I work on last week?
- Finished the frontend for [the gno version of disperse](https://github.com/agherasie/gno-disperse)
- Tried setting up tests for the smart contract but ran into some problems

#### What will I focus on this week?
- Finishing disperse with @lennyvong
- Starting a new project / polishing [gno-forms](https://github.com/agherasie/gno-forms) if considered relevant

#### Do I have any blockers?
- Testing disperse, as explained in the signal group, there is a problem with the realm having an amount of GNOT from the beginning in the testing scope

---

### Reza (@RezaRahemtola)

#### What did I work on last week?
- Tested some things out locally with GRC20 and Coins to prepare the Aave project
- Also looked at Uniswap v2 because some people (friends from Kiln) recommanded me to build something like this on Gno first before going for something really big and "hard" as an Aave-like project
- Not related to Gno but won a prize at ETHGlobal Brussels, maybe it could be interesting to build something like this on Gno ? https://ethglobal.com/showcase/near-ai-assistant-jmwgf

#### What will I focus on this week?
- Start (finally) the MVP of Aave
- Or maybe the AI stuff, like a chatbot using a model deployed on a decentralized cloud and fed with the markdown of the Gno docs in realtime (like updated when the docs are updated) to help devs and users onboard and find information ?

#### Do I have any blockers?
- Nope

---

### Mohammed (@Molaryy)

#### What did I work on last week?
- I finished implementing the GRC721 for our [openocean](https://github.com/Molaryy/openocean) project
- I participated to the ethglobal and we did a project with the Inco network, the project was about the Liar's Dice, so I think once there will be an oracle in gno to use random I will try to reproduce it.

#### What will I focus on this week?
- This week I will sync the frontend with the realm
- I will start a new project

#### Do I have any blockers?
- No

**Comment by leohhhn** [comment]
22-26 July 2024

### Alex (@agherasie)

#### What did I work on last week?
- Connecting the frontend to the smart contract and ipfs (via pinata) for [openocean](https://github.com/Molaryy/openocean) with @Molaryy

#### What will I focus on this week?
- Finishing the openocean implementation (deploying on test4 and opening a PR)

#### Do I have any blockers?
- Not exactly, just waiting for feedback on my PRs:
    - [disperse](https://github.com/gnolang/gno/pull/2613)
    - [forms](https://github.com/gnolang/gno/pull/2604)
    - [readme quick link](https://github.com/gnolang/gno/pull/2461)


---
### Lenny (@lennyvohn)

#### What did I work on last week?
- I've contacted the Gnorkle developer to talk about his Oracle and possible improvements (deelawn). He sent me quite a few things to test, so I'll have to get back to him by mid-week.
- Going around Gnorkle and testing the things deelawn sent me to see how his implementation works and especially to see the limitations.
- Started thinking about a project where I'll need an oracle to test Gnorkle in parallel. It's not a big project, but enough to be able to test the Oracle in its entirety.

#### What will I focus on this week?
-I'm going to focus on Gnorkle this week, producing a dapp that needs off-chain data.
-Note the limitations of the current Gnorkle implementation and then discuss with the developer how to improve it

#### Do I have any blockers?
- Some about Gnorkle but Deelawn gives quick answers.

---

### Reza (@RezaRahemtola)

#### What did I work on last week?
Almost fully on Adena this time
- Built a great MVP on a TS SDK for Adena: https://github.com/RezaRahemtola/adena-sdk-ts
- Tested it & used it in gnovox to make sure it works correctly
- Raised an issue on a response improvement on Adena: https://github.com/onbloc/adena-wallet/issues/538
- Made a PR to fix an earlier proposition I made: https://github.com/onbloc/adena-wallet/pull/539

#### What will I focus on this week?
- Enhancing gnovox with these new methods / accessibility
- Maybe publish the SDK / see with the onbloc team the next steps and enhance my basic SDK
- Continue to go through this duality of building a project (gnovox) and enhancing the tools I use in the meantime with the issues / suggestions I get as a "dApp developer"

#### Do I have any blockers?
- Nope, just waiting on a few feedbacks here:
  - [AI assistant](https://github.com/gnolang/gno/issues/2611)
  - 2 small docs PR waiting for a month:
      - https://github.com/gnolang/gno/pull/2427
      - https://github.com/gnolang/gno/pull/2419

---

### Mohammed (@Molaryy)

#### What did I work on last week?
- We finally finished [openocean](https://github.com/Molaryy/openocean) with [Alex](https://github.com/agherasie)
- I started thinking about how to implement [gitland](https://github.com/Molaryy/gitland), Gitcoin like using the GNO technology
- I had a call with some Rouen students about implementing the oracle and I went to ask the creator of [gnorkle](https://github.com/gnolang/gno/tree/master/examples/gno.land/p/demo/gnorkle) because he already implemented something similar


#### What will I focus on this week?
- I will continue gitland
- I will do more research about the gnorkel and the VRFs

#### Do I have any blockers?

- Nope

---

### zak (@kazai777)

#### What did I work on last week?
- Start of development of a dApp to offer tasks in return payment

#### What will I focus on this week?
- Continue development

#### Do I have any blockers?
- No

**Comment by leohhhn** [comment]
29 July - 2 August 2024

### Lenny (@GH_username)

#### What did I work on last week?
- Oracle: Started a Dapp to use Gnorkle and takes some notes about what could be improve. A sports bettings platform with @agherasie. This project is essently to test gnorkle and give some feedbacks but it can be a cool example project.
- Discuss about a bigger project with @reza. A multi-sig adapted for investement and group trading by implement shared rewards. Kind a mix between disperse and Safe. It could be really interessting as Gno has not a complex multi-sig for now. 

#### What will I focus on this week?
- Continue to develop the sports bettings plateform to discover Gnorkle and its features.
- Start to setup the project about a complex multi-sig I really think it could be great.

#### Do I have any blockers?
- No

---

### Alex (@agherasie)

#### What did I work on last week?
- Adding final features to openocean with @molaryy
    - buying nfts
    - starring collections
    - stats
    - etc
- Addressing @gfanton 's comments on r/forms and r/disperse
- Migrating to filetests for r/disperse

#### What will I focus on this week?
- I would like to help @RezaRahemtola on the adena sdk and also integrate it to my existing projects
- Deploying [openocean](https://github.com/Molaryy/openocean) on test4 (hopefully early this week)

#### Do I have any blockers?
- Not exactly

---


### Reza (@RezaRahemtola)

#### What did I work on last week?
- Cleaned / enhanced a few details on the Adena SDK and made a [PR](https://github.com/onbloc/adena-wallet-sdk/pull/1) to the official repo on which we'll work in the future on the SDK to publish it
- As suggested by Guilhem started working on a PoC of the Gno AI assistant (based on the suggestion issue I previously opened). The markdown feeding part is a bit challenging (have to use 2 models, one for function calls and one with a big context length) but I should have something basic ready soon (fully decentralized, deployed and accessible, maybe not fed in realtime with the docs yet)
- Discussed an idea of multisig revenue sharing with Lenny

#### What will I focus on this week?
- Finish my MVP of the AI assistant
- Once the PR of adena-sdk is merged, see with the Onbloc team / Milos how I can work with them to improve it
- Maybe start the multisig idea with Lenny, or continue or gnovox (which kinda became my backup solution in case I don't really know what to do)

#### Do I have any blockers?
- Nope

---

### Mohammed (@Molaryy)

#### What did I work on last week?
- Basically did a big refactoring on [openocean](https://github.com/Molaryy/openocean)
- Started the gitland (inspiration of [gitcoin]()https://www.gitcoin.co/) webapp and also the realm.
    - Thought about the passeport integration
        - You get more points if you connect more of your third-party accounts (guthub, google, etc) 

#### What will I focus on this week?
- We still have a few weeks left for the student program, so I will try to finish gitland before that, even if I will still working on it after that
- Make an issue about gno test, can't add global variables.

#### Do I have any blockers?
- No

**Comment by leohhhn** [comment]
Aug 5th - 9th 2024

### Reza (@RezaRahemtola)

#### What did I work on last week?
- AI Assistant (was pretty challenging to build but it works, with RAG on chunks of the Markdown documentation)
(!![image](https://hackmd.io/_uploads/ryiHPqw5C.png))
Will open source the repo after I cleaned up a few things with a tutorial so anyone can run it locally
- Adena SDK is merged! Not yet deployed on NPM I think but merged it with the onbloc team and fixed some bugs spotted by Alex

#### What will I focus on this week?
- Work with Mohammed on gitland, can be great to finish a basic version of this project before the end of the project
- Discuss with Milos and jinoosss to contribute further on the SDK

#### Do I have any blockers?
- No

---
### Alex (@agherasie)

#### What did I work on last week?
- Worked on disperse (adressing the final comments by to hopefully get the PR merged)
- Deployed openocean on test4 and netlify (for the frontend) and shared it with the team on signal
- Tried looking at gno native, ran into some bugs but would maybe like to do something with it this week
- Partially switched openocean to @RezaRahemtola 's adena sdk

#### What will I focus on this week?
- Helping @lennyvong with the frontend of his betting project
- Try and get pending PRs merged as always
- Maybe try and do something with gno native ? 

#### Do I have any blockers?
- waiting for the adena sdk to be uploaded as an npm package so I can migrate all my code to using it
- bugs when trying gno native examples, will probably open an issue
---


### Lenny (@GH_username)

#### What did I work on last week?
- Gnobet (https://github.com/lennyvong/gnobet):
I focused on developing the realm of my sports betting Dapp. I explored almost all of Gnorkle and I understood how it worked and I was able to test different things.
I started to think about the structure of the oracle off-chain agent I would need and began to test it. @agherasie is helping me for the front.
- Gnorkle:
I don't think Gnorkle is 100% suited to my needs, so I've sent some feedback to Dylan and I'm waiting for his answers. I'd like to continue developing the other feeds, but only the static feed is implemented, so I'm waiting to hear about the problems he's encountered.

#### What will I focus on this week?
- Finish a first version of the platform by be able to bet on the pre-match so I need to finish the realm and the off-chain agent in go and connect with the front. And setting up the suite for be able to bet "in-game".
- Discuss with dylan if it's possible to resume the developpement of Gnorkle because the feed I need is not the one implemented ("Continiously").

#### Do I have any blockers?
- about Gnorkle

---
### Theo (@DIGIX666)
### What did I work on last week?
- I worked on : 
    - issue(`TotalCoin`) [2662](https://github.com/gnolang/gno/issues/2662) -> PR [2670](https://github.com/gnolang/gno/pull/2670)
    - issue(`-race flag`) [2097](https://github.com/gnolang/gno/issues/2097) -> PR [2663](https://github.com/gnolang/gno/pull/2663)
### What will I focus on this week?
- I focus to :
    - issue [2097](https://github.com/gnolang/gno/issues/2097) -> PR [2663](https://github.com/gnolang/gno/pull/2663)
    - PR [2076](github.com/gnolang/gno/pull/2076) update my UUID package for use `math/rand` because `math/rand` is merged ([2455](https://github.com/gnolang/gno/pull/2455))

### Do I have any blockers?
- Nope

---

### Malek (@MalekLahbib)

#### What did I work on last week?
- trying to understand gnovm and watch the last commits to know what's new

#### What will I focus on this week?
- same thing maybe

#### Do I have any blockers?
- gnovm: where do I start? need guidance
- Maybe focus on other things?

---

### Mohammed (@Molaryy)

#### What did I work on last week?
- Opened this [pr](https://github.com/gnolang/gno/pull/2678) to propose [Openocean](https://github.com/Molaryy/openocean) into the examples folder, it would be good to get some comments
- When I was trying to do the pr I wanted to fix some types so I found a problem, this is the [issue](https://github.com/gnolang/gno/issues/2665)
- I worked on the frontend of [gitland](https://github.com/Molaryy/gitland) and also on the realms
- Still exploring about the oracles for the dice game

#### What will I focus on this week?
- Finish the realm of gitland for tuesday

#### Do I have any blockers?
- Nop

**Comment by leohhhn** [comment]
Aug 12th - 21st 

### Alex (@agherasie)

#### What did I work on last week?
- Got sick (in August yes) so I couldn't do as much as I hoped
- Made a first version of the gnobet frontend for @lennyvong 's project
- Fixed 's new feedback on disperse
- Read about gRPC, tried to understand how gno-js-client works and made my own GetBalance call in python

#### What will I focus on this week?
- Finishing the gnobet frontend
- Hopefully merging gno forms and disperse
- Moving to South Korea !

#### Do I have any blockers?
- Getting sick was definitely a blocker but I'm okay now

---


### Lenny (@lennyvong)

#### What did I work on last week?
Gnobet (https://github.com/lennyvong/gnobet/pull/2):
- The realm is almost finished for a first simple version and the functions to communicate with the front.
- the off-chain-agent is also functional and communicates correctly with the realm thanks to Gnorkle and dylan. I came up with the idea of using events instead of calling a function to retrieve the requests to be processed. It works fine, and it's even better. It's more suited to my use of gnorkle

Gnorkle (https://github.com/lennyvong/gno): 
- I needed to implement a new feed (“continious”) to be able to update the state continuously. So it's implemented on a fork but I need to get some feedbacks from dylan if he saw such an implementation. But currently this implementation suits my use better than the “static” one.

#### What will I focus on this week?
- Finish gnobet for a first implementation with frontend.
- See with dylan if that is what he wanted for Gnorkle feed.

#### Do I have any blockers?
- not really


---

### Reza (@RezaRahemtola)

#### What did I work on last week?
- Cleaned up and made public the AI assistant: https://github.com/RezaRahemtola/gno-ai-assistant
- Made some PRs on Adena to add a CheckEstablish method:
    - https://github.com/onbloc/adena-wallet/pull/539
    - https://github.com/onbloc/adena-wallet-sdk/pull/2
  and made a local PoC of it
- Worked with @Molaryy on gitland, a bit more complicated that expected but we still have a small demo ;)

#### What will I focus on this week?
- Fix the comments on my PRs to make sure they get merged
- Fix the bugs of gitland if possible to make it works well
- Taking my plane to Korea tomorrow ;)

#### Do I have any blockers?
- Float/int conversion error on gitland

---

### Mohammed (@Molaryy)

#### What did I work on last week?
- Did some research about the oracles with Lenny and he found out how to do it
- I mainly worked on [gitland](https://github.com/Molaryy/gitland) with Reza, he updated the frontend to make it look prettier and also add the web3 implementations and I updated the realm so now it works perfectly and I just have 2 little bugs to fix

#### What will I focus on this week?
- Fix the little bugs and we will finish the project

#### Do I have any blockers?
- I got an error while trying to do this:
```go
int64(float64(coins.AmountOf("ugnot") * 0.02))
```
---

### Theo (@DIGIX666)
### What did I work on last week?
- I worked on : 
    - PR(`UUID`) [2076](https://github.com/gnolang/gno/pull/2076) -> merged `p/demo/entropy`
    - issue(`-race flag`) [2097](https://github.com/gnolang/gno/issues/2097) -> PR [2663](https://github.com/gnolang/gno/pull/2663)
### What will I focus on this week?
- I focus to :
    - issue(`-race flag`) [2097](https://github.com/gnolang/gno/issues/2097) -> PR [2663](https://github.com/gnolang/gno/pull/2663)
    - PR [2076](github.com/gnolang/gno/pull/2076) finish a review my UUID package 
    - PR [2307](https://github.com/gnolang/gno/pull/2307) finish a review `accescontrol` and `timelock` package

### Do I have any blockers?
- Nope

**Comment by leohhhn** [comment]
## New SCP batch - RAF Cohort

11.12.2024-18.12.2024

### Ivan (@Ursulovic)

#### What did I work on last week?
- Last week I trying to get most familiar with the ecosystem most as I could, I was reading documentation, looking at other member's PRs and issues, I also watched Leon's live coading session. When I figured I had enoguh knowlege, I started writting my Home Realm. It's main functionality was displaying pseudo-random selected blockchain "wisdom" aswell as displaying some info about me. While it still needs some polishin I have md PR for it.

#### What will I focus on this week?
- I will continue exploring realms in demo
- I will try to port my Ethereum project to GnoLand(Decentralized torrent tracker service)
- I will continue exploring ecosystem 


#### Do I have any blockers?
- Not blockers, but I feel I have a lot more to learn

--- 

### Stefan (@stefann-01)

#### What did I work on last week?
- Submitted a FOMO3D game implementation and some tests, it's not finished yet. [3344](https://github.com/gnolang/gno/pull/3344)
- Proposed an extension to the `Ownable` contract to support multiple superuser addresses. [3347](https://github.com/gnolang/gno/issues/3347)
- Did some research about gnovm and function injections after running into potential bug with starting local chain.

#### What will I focus on this week?
- Finishing the FOMO3D game, and maybe making frontend part to make user expirience even better. [3344](https://github.com/gnolang/gno/pull/3344)
- Working on the `Ownable` extension mentioned above. [3347](https://github.com/gnolang/gno/issues/3347)
- Working on some of the other proposals mentioned in blockers.

#### Do I have any blockers?
- Pending responses on updating documentation and examples related to PrevRealm and OrigCaller. [2923](https://github.com/gnolang/gno/issues/2923)
- Awaiting feedback on the proposal for a fine-tuned ChatGPT plugin tailored for Gno developers to assist with development challenges. [2950](https://github.com/gnolang/hackerspace/issues/104)

---

### Matija Marjanovic (@matijamarjanovic)

#### What did I work on last week?
- Reformated my homerealm to have support for markdown, got approved
- I have spent a lot of time exploring and trying to use tokens in small and insignificant 'apps' and tried to make meme coins of my own. You can find these [here](https://github.com/matijamarjanovic/9gno) and [here](https://github.com/matijamarjanovic/rafOnChainStudentRegistry). 

#### What will I focus on this week?
- This week I have started by trying to improve the grc package, but have soon found it there is not much room for improvement: I will keep focusing on understanding the usage of tokens by building an app that might actually be useful.
- Exploring the BTree and it's usages

#### Do I have any blockers?
- none

---

### Jovan (@JJOptimist)

#### What did I work on last week?
- Last week I was mostly reading through docs and exploring already existing issues at first. Then i moved to experimenting
with GRC20 and started developing my homerealm(still exploring possibilities what can you display) 

#### What will I focus on this week?
- For this week i will focus on GRC721 and development of my Event Ticketing system using GRC721 and integration into TG Mini App. I also got interested in gnoweb while developing my homerealm so i will be exploring that part too and looking to improve it. I will link everything soon.

#### Do I have any blockers?
-

**Comment by leohhhn** [comment]
19.12.2024-26.12.2024

### Matija Marjanovic (@matijamarjanovic)

#### What did I work on last week?
Last week I have encountered a lot of setbacks in terms of realms  
I have been working on. For instace I have started building a reddit-like
social media app on chain until I was reminded that "boards" exist... Although,
my work was not in vain as I have gotten to understand some G(n)o concepts  
I was unfamiliar with. 
I continued exploring the monorepo and have starting building a platform to store user notes on chain (draft PR is open).

#### What will I focus on this week?
I will continue working on the notes app and trying to think of new usefull apps and packages on chain.

#### Do I have any blockers?
Only in terms of creativity (I'm open to all suggestions on what I could contribute to).

--- 

### Stefan Nikolić (@stefann-01)

#### What did I work on last week?
Last week, I reached a point where my lack of knowledge in Go and my difficulty differentiating pure Go from Gno (both in libraries and design choices) started slowing me down. I spent a lot of time researching and understanding these differences rather than actively developing. Despite this, I almost finished the Fomo3D game, introduced some improvements, and implemented new features: [PR gnolang/gno#3344](https://github.com/gnolang/gno/pull/3344).

#### What will I focus on this week?
This week, I plan to integrate NFTs into the game so that an NFT is minted for every round winner. Additionally, I hope to collaborate with @n2p5 on developing some Ownable extensions: [Issue gnolang/gno#3347](https://github.com/gnolang/gno/issues/3347).

#### Do I have any blockers?
Nope.

---

### Ivan Ursulovic (@ursulovic)

#### What did I work on last week?
Last week, I finished my home realm and made a [PR](https://github.com/gnolang/gno/pull/3354). The code is good, but the home realm is not unique enough to be published. I realized that I still have significant knowledge gaps in Go, so I continued learning about the language and its best practices. I also created a specification for the future proposal I mentioned last week (Decentralized Torrent Tracker Service), which you can read here: [link](https://cultured-whitefish-825.notion.site/Decentralized-torrent-tracker-service-16b18fd4621f800a80d0e2824bdb4f4f?pvs=4). I’ve continued exploring the monorepo.

#### What will I focus on this week?
I plan to create a new, more unique home realm. Additionally, I’d like to start working on the Decentralized Torrent Tracker Service if you find the project interesting. I will continue exploring both Gno and Go

#### Do I have any blockers?
None at the moment.

---

### Jovan Jovanovic (@JJOptimist)

#### What did I work on last week?
I worked on my homerealm, i brainstormed different ideas of specific things to add to my realm it will be more challenging but also more interesting. I laid the foundation for my Event ticketing system but i will pause that for a moment until i try something simpler to get better with GRC-721. I went through existing gnoweb issues and tried to see current problems and ways to improve it.


#### What will I focus on this week?
Resolving current issues within my homerealm and adding some interesting feature to it. Go back through awesome-gno and get better at fundamentals. After that head back to my Event ticketing project and work on it. Try to resolve an issue in gnoweb

#### Do I have any blockers?
/

**Comment by leohhhn** [comment]
Jan 8th 2025

### Stefan Nikolic (@stefann-01)

#### What did I work on last week?

- Refactor the old home realm to integrate with the new Gnoweb, plus adding some additional features. [PR gnolang/gno#3456](https://github.com/gnolang/gno/pull/3456)
- Collaborated with @matijmarjanovic to create the first version of the Gno Developer Assistant ChatGPT plugin. [Issue gnolang/hackerspace#104](https://github.com/gnolang/hackerspace/issues/104)
- Began implementing the NFT feature for the FOMO3D game. [PR gnolang/gno#3344](https://github.com/gnolang/gno/pull/3344)

#### What will I focus on this week?

- Completing the NFT feature implementation for the FOMO3D game.
- Initiating work on lending protocol that uses DEX-based price discovery for loans, eliminating the need for external oracles.

#### Do I have any blockers?

- No

---

### Matija Marjanovic (@matijamarjanovic)

#### What did I work on last week?

- Contributed to the development of the initial version of ChatGPT plugin designed to enhance Gno documentation accessibility and usability [see issue](https://github.com/gnolang/hackerspace/issues/104)

#### What will I focus on this week?

- Initiating the design and implementation of a decentralized exchange

#### Do I have any blockers?

---

### Ivan Ursulovic (@ursulovic)

#### What did I work on last week?

- I have almost finished on my proposal for decentralised torrent tracker service [Link](https://cultured-whitefish-825.notion.site/Decentralized-torrent-tracker-service-16b18fd4621f800a80d0e2824bdb4f4f?pvs=4), I did not only explore Gno, but general system design. I have created Electron app prototype, and began writting live seeder/leecher tracker in Go using anacrolix/torrent library


#### What will I focus on this week?

- I will post final Decnetralised torrent tracker service speicifations
- Continue learning

#### Do I have any blockers?

-None yet

--- 

### Jovan Jovanovic (@JJOptimist)

#### What did I work on last week?

- I have been fixing issues on my home realm pointed by Leon such as using existing packages/libs like ownable to be more efficient. I experimented with some CSS possibilities trying to make my homerealm unique, after i develop what i envisioned i will put it again for review. I've been exploring gno.land/pkg/gnoweb/components and filling the knowledge gaps i had with gno and go lang.


#### What will I focus on this week?

 Last week i paused my project for Event ticketing system until i fill the gap in knowledge and i will continue it now. Add something more creative to my homepage. If possible colab with Alex (saw hes been working recently on gnoweb issues)on something with gnoweb since im interested in that part of gno.land also.

#### Do I have any blockers?

-None yet
